### PR TITLE
added Task Desc and Metrics used for Text-To-Image

### PR DESCRIPTION
**Pull request by samyak1999**

### CODE REVIEW

**Comment by merveenoyan** [comment]
@samyak1999 Hello and thanks a lot for working on this! 
The task description and metrics belong to `data.ts` you can use this [template](https://github.com/huggingface/hub-docs/blob/main/tasks/src/placeholder/data.ts) and fill the information there. 🙂 
If you have any questions please let me know!

File: tasks/src/text-to-image/data.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import type { TaskDataCustom } from "../Types";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+		{
+			// TODO write proper description
+			description: "Caltech-UCSD Birds-200-2011 is an extended version of the CUB-200 dataset, with roughly double the number of images per class and new part location annotations.",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:7]: Can you add datasets that are on Hub and are rather canonical? Couple of examples you could pick from: 
https://huggingface.co/datasets/red_caps
https://huggingface.co/datasets/conceptual_captions
https://huggingface.co/datasets/conceptual_12m
they're easier to load and are generic 🙂

File: tasks/src/text-to-image/data.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		}
+
+	],
+	demo: {
+		inputs: [],
+		outputs: []
+	},
+	metrics: [
+		{
+			description: "The inception score involves using a pre-trained deep learning neural network model for image classification to classify the generated images.",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:52]: What does IS measure?

File: tasks/src/text-to-image/data.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			id:          "IS",
+		},
+		{
+			description: "The Fréchet inception distance (FID) is a metric used to assess the quality of images created by a generative model, like a generative adversarial network (GAN).",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:56]:
```suggestion
			description: "The Fréchet inception distance (FID) is a metric used to assess the quality of images created by a generative model.",
```
Can you specify how the quality is measured?

File: tasks/src/text-to-image/data.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+			id:          "FID",
+		},
+		{
+			description: "R-precision is defined as the proportion of the top-R retrieved documents that are relevant, where R is the number of relevant documents for the current query.",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:60]: What is the document in this case? And also relevancy?

File: tasks/src/text-to-image/data.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+			id:          "R-precision",
+		},
+		{
+			description: "L2 loss, also known as Squared Error Loss, is the squared difference between a prediction and the actual value, calculated for each example in a dataset",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:64]: How does this apply for this task specifically?

File: tasks/src/text-to-image/data.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+			id:          "L2",
+		},
+		{
+			description:"'Learned Perceptual Image Patch Similarity' evaluates the distance between image patches. Higher means further/more different. Lower means more similar.",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:68]: What does distance mean in this context? How is it calculated?

File: tasks/src/text-to-image/data.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+	],
+	models: [
+		{
+			// TO DO: write description
+			description: '',
+			id:          ''
+		},
+		
+	],
+	summary:      "Deep learning research has achieved several significant technological advances in the domains of Computer Vision (CV) and Natural Language Processing (NLP) over the past few decades. In these hitherto distinct domains, researchers now seem interested in fusing semantic and visual information. Numerous studies have been done on text-to-image synthesis methods, which convert the textual description (keywords or sentences) that is input into realistic visuals",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:81]:
```suggestion
	summary:      "Deep learning research has achieved several significant technological advances in the domains of Computer Vision (CV) and Natural Language Processing (NLP) over the past few decades. In these hitherto distinct domains, researchers now seem interested in fusing semantic and visual information. Numerous studies have been done on text-to-image synthesis methods, which convert the textual description (keywords or sentences) that is input into realistic visuals",
```
Can you make this more focused and simple for text-to-image task itself? Doesn't have to be long 🙂

File: tasks/src/text-to-image/data.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+	widgetModels: [],
+	youtubeId:    "SVcsDDABEkM",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:83]:
```suggestion
	youtubeId:    "",
```
I think for video we would like to keep it simplistic like other pages, we don't have any video.

**Review by merveenoyan — COMMENTED** [review]
Overall LGTM, we just need to simplify the language a little and reduce ambiguities 🙂 Think of this as a wikipedia page made to get people informed enough to use or assess a model 🙂

**Comment by samyak1999** [comment]
Thanks @merveenoyan for pointing out these. I will resolve these ASAP and try to make it more simplistic

File: tasks/src/text-to-image/data.ts (hunk 1 above)
Comment (merveenoyan) [reply tasks/src/text-to-image/data.ts:7]: We usually try to have 2-3 models so I'd be in favor of removing 🙂

File: tasks/src/text-to-image/data.ts (hunk 2 above)
Comment (samyak1999) [reply tasks/src/text-to-image/data.ts:52]: The score seeks to capture two properties of a collection of generated images:
- Image Quality - Do images look like a specific object?
- Image Diversity-  Is a wide range of objects generated?

The inception score is calculated by first using a pre-trained Inception v3 model to predict the class probabilities for each generated image.

Comment (merveenoyan) [reply tasks/src/text-to-image/data.ts:52]: I'm not asking to learn but rather asking from a reader's perspective, can you explain it well?

**Comment by merveenoyan** [comment]
Partially fixes #364

File: tasks/src/text-to-image/data.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	
+		{
+			description:"'Learned Perceptual Image Patch Similarity'is used to judge the perceptual similarity between two images. A low LPIPS score means that image patches are perceptual similar. ",
+			id:			 "LPIPS"
+		},
+
+	],
+	models: [
+		{
+			// TO DO: write description
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:43]: Imagen is closed source and there's no way to infer with it, so we don't want to have that given we want these pages to have open-source everything.

File: tasks/src/text-to-image/data.ts [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+			description: 'Imagen, a text-to-image diffusion model with an unprecedented degree of photorealism and a deep level of language understanding.',
+			id:          'imagen'
+		},
+		
+	],
+	summary:      "Because of the advancement of Deep Learning technologies, we can generate anything we want that also invloves automatic synthesis of realistic images from text or text-to-image synthesis. ",
```
Comment (merveenoyan) [inline tasks/src/text-to-image/data.ts:49]: A contributor already contributed a definition, you can see here: https://github.com/huggingface/hub-docs/blob/main/tasks/src/text-to-image/data.ts

**Review by merveenoyan — COMMENTED** [review]
We've already had a contributor contributing couple of these sections and it overlaps with your PR, but we can have the metrics given they haven't been contributed (you can see here https://github.com/huggingface/hub-docs/blob/main/tasks/src/text-to-image/data.ts) and I'd encourage you to merge main branch into this branch and solve merge conflicts before making next round of changes (let me know if you need help with it 🙂)

**Comment by merveenoyan** [comment]
@samyak1999 do you need extra help with this?

**Comment by jayant-yadav** [comment]
https://github.com/huggingface/hub-docs/pull/801 is open for review.

**Comment by osanseviero** [comment]
Closing this as it was superseded by https://github.com/huggingface/hub-docs/pull/801

**Pull request closed without merging**